src/OrbitBase/CanceledOrTest.cpp

```diff
@@ -14,29 +14,45 @@
 
 namespace orbit_base {
 
-TEST(CanceledOr, Construct) {
+TEST(CanceledOrVoid, NotCanceled) {
   CanceledOr<void> void_or_canceled{};
   EXPECT_TRUE(void_or_canceled.HasValue());
   EXPECT_FALSE(void_or_canceled.IsCanceled());
+}
 
-  void_or_canceled = Canceled{};
+TEST(CanceledOrVoid, Canceled) {
+  CanceledOr<void> void_or_canceled{Canceled{}};
   EXPECT_FALSE(void_or_canceled.HasValue());
   EXPECT_TRUE(void_or_canceled.IsCanceled());
+}
 
+TEST(CanceledOrInt, NotCanceled) {
   CanceledOr<int> int_or_canceled{42};
   EXPECT_TRUE(int_or_canceled.HasValue());
   EXPECT_FALSE(int_or_canceled.IsCanceled());
   EXPECT_EQ(int_or_canceled.GetValue(), 42);
+  EXPECT_EQ(static_cast<const CanceledOr<int>&>(int_or_canceled).GetValue(), 42);
+  // NOLINTNEXTLINE(performance-move-const-arg)
+  EXPECT_EQ(std::move(int_or_canceled).GetValue(), 42);
```

Comment:
What's the point of the check? If there was only `GetValue() const &`, it would also work. I suggest to use a static assert to verify the return type.